File: igVortex/Python/src/velocityPlot/igplotVelocity.py
from matplotlib import scale
import numpy as np
import math
import matplotlib.pyplot as plt
import src.globals as g
from src.velocityPlot.igVELF import igVELF
from src.velocityPlot.igVELOCITYF import igVELOCITYF


def igplotVelocity(istep, ZV, ZW, a, GAMA, m, GAMAw, iGAMAw, U, V, alp, l, h, dalp, dl, dh):
    """
    Plot Velocity field

    Input:
    * istep: iteration step
    * ZV: bound vortex location
    * ZW: wake vortex location
    * a: rotation axis offset.
    * GAMA: bound vortex
    * m: # of bound vortices
    * GAMAw: wake vortices
    * iGAMAw: # of wake vortices
    * U, V: free flow velocity
    * alp: airfoil rotation
    * l, h: airfoil translation
    * dalp, dl, dh: airfoil velocity components

    Global Variables:
    * ZETA: mesh points matrix
    * zavoid: avoid vortex points for velocity calculation (slower)
    * folder: fig folder path
    * svCont, wvCont: speed contour plot velocity range specifier
    * vpFreq: frequency of velocity field plot.
    * ivCont: switch for use of svCont and wvCont: use them if ivCont == 1.
    """

    # Airfoil 9/10/2018
    XPLTF = np.real(ZV)
    YPLTF = np.imag(ZV)

    # Plot the velocity field, every vpFreq seps.
    if istep % g.vpFreq == 0:
        # Calculate the velocity field.
        ROT = np.exp(-1j * alp)
        RZETA = (g.ZETA + a) * ROT

        X = np.real(RZETA) + l
        Y = np.imag(RZETA) + h
        Z = X + 1j * Y

        if g.zavoid == 1:
            # Skip source points that coincides with the obseration points.
            # (slower)

            VVspace = igVELF(Z, ZV, ZW, GAMA, m, GAMAw,
                             iGAMAw, U, V, alp, dalp, dl, dh)
        else:
            VVspace = igVELOCITYF(Z, ZV, ZW, a, GAMA, m, GAMAw,
                                  iGAMAw, U, V, alp, dalp, dl, dh)

        # Plot the velocity field in the space-fixed system.

        U = np.real(VVspace)
        V = np.imag(VVspace)
        S = np.sqrt(U * U + V * V)
        S = np.reshape(
            S, (int(math.sqrt(S.shape[0])), int(math.sqrt(S.shape[0]))))

        plt.quiver(X, Y, U, V)
        plt.plot(XPLTF, YPLTF, '-b')
        plt.savefig(f"{g.folder}velocity/spaceVelocity_{istep}.png")
        plt.clf()

        if g.ivCont == 1:
            plt.contour(X, Y, S, g.svCont)
            plt.contourf(X, Y, S, g.svCont)
        else:
            plt.contour(X, Y, S)
            plt.contourf(X, Y, S)

        plt.colorbar()

        plt.plot(XPLTF, YPLTF, '-b', linewidth='4')
        plt.savefig(f"{g.folder}velocity/spaceSpeed_{istep}.png")
        plt.clf()

File: 2D Simulation/MATLAB/velocityPlot/igplotVelocity.m
function [  ] = igplotVelocity( istep,ZV,ZW, a,GAMA,m,GAMAw,iGAMAw,alp,l,h)
%Plot velocity field
% 
%INPUT
% istep     iteration step
% ZV        bound vortex location
% ZW        wake vortex location
% a         rotation axis offset
% GAMA      bound vortex
% m         # of bound vortices
% GAMAw     wake vortices
% iGAMAw    # of wake vortices
% U,V       free flow velocity
% alp       airfoil rotation
% l h       airfoil translation
% dalp, dl, dh  airfoil velocity components
global ZETA  zavoid folder svCont wvCont vpFreq ivCont
% ZETA      mesh points matrix
% zavoid    avoid vortex poits for velocity calculation (slower)
% folder    fig folder path
% svCont, wvCont    speed contour plot velocity range specifier
% vpFreq    frequency of velocity field plot
% ivCont    swich for use of svCpont and wvCont: use them if ivCont ==1

%Airfoil 9/10/2018
    XPLTF  =real(ZV);
    YPLTF  =imag(ZV);

%Plot the velocity field, every vpFreq seps
if rem(istep,vpFreq) == 0
    %Calculate velocity field
    ROT=exp(-1i*alp);
    RZETA=(ZETA+a)*ROT;
    X  =real(RZETA) + l;
    Y  =imag(RZETA) + h;
    Z=complex(X,Y);
    if zavoid == 1 %skip source points that coincides with the observation points (slower)
        [VVspace]=igVELF(Z,ZV,ZW,GAMA,m,GAMAw,iGAMAw);
    else
        [VVspace]=igVELOCITYF(Z,ZV,ZW,GAMA,m,GAMAw,iGAMAw);
    end
    %Plot velocity field in the space-fixed system
  
    
    U  =real(VVspace);
    V  =imag(VVspace);
    S  =sqrt(U.*U+V.*V);
    quiver(X,Y,U,V)
    hold on; %9/10/2018
    plot(XPLTF, YPLTF, '-k'); %9/10/2018
    hold off; %9/10/2018
    print('-dbmp',[folder 'velocity/' 'spaceVelocity_' num2str(istep) '.bmp']);
    close;
    if ivCont == 1
        contourf(X,Y,S,svCont);
    else
        contourf(X,Y,S);
    end
    colorbar;
    hold on; %9/10/2018
    plot(XPLTF, YPLTF, '-k'); %9/10/2018
    hold off; %9/10/2018
    print('-dbmp',[folder 'velocity/' 'spaceSpeed_' num2str(istep) '.bmp']);
    close;
    
    
end
end

